File: VB/WebSite/topgrid.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls

Partial Public Class topgrid
	Inherits System.Web.UI.UserControl
	Private Const RowCount As Int32 = 230000

	Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		Dim tbl As New DataTable()

		Dim col As New DataColumn("Id", GetType(Int32))
		tbl.Columns.Add(col)
		tbl.PrimaryKey = New DataColumn() { col }

		col = New DataColumn("Name", GetType(String))
		tbl.Columns.Add(col)

		col = New DataColumn("Text", GetType(String))
		tbl.Columns.Add(col)

		For i As Integer = 0 To RowCount - 1
			tbl.Rows.Add(New Object() { i, String.Format("Name{0}", i), String.Format("Test{0}", i) })
		Next i

		grid1.DataSource = tbl
		grid1.DataBind()
	End Sub
End Class

File: CS/WebSite/topgrid.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

public partial class topgrid : System.Web.UI.UserControl
{
    const Int32 RowCount = 230000; 

    protected void Page_Load(object sender, EventArgs e)
    {
        DataTable tbl = new DataTable();
        
        DataColumn col = new DataColumn("Id", typeof(Int32));
        tbl.Columns.Add(col);
        tbl.PrimaryKey = new DataColumn[] { col };
        
        col = new DataColumn("Name", typeof(String));
        tbl.Columns.Add(col);

        col = new DataColumn("Text", typeof(String));
        tbl.Columns.Add(col);

        for (int i = 0; i < RowCount; i++)
            tbl.Rows.Add(new Object[] { i, String.Format("Name{0}", i), String.Format("Test{0}", i) });

        grid1.DataSource = tbl;
        grid1.DataBind();
    }
}
